site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx

```diff
@@ -47,13 +48,19 @@ export interface CategoryWithPercent extends Category {
 const calculateFootprintPercent = (
   total: number,
   categories: Category[]
-): CategoryWithPercent[] => {
-  return categories.map((category, index) => ({
-    ...category,
-    percent: (category.quantity / total) * 100,
-    fill: COLOR_PALETTE[index],
-  }));
-};
+): CategoryWithPercent[] =>
+  categories
+    /**Sort the categories from largest to smallest for the correct color order */
+    .sort((cat1, cat2) => cat2.quantity - cat1.quantity)
```

Comment:
Ah I see, @theprofessora does the order of colour in the graph matter? From the designs it looks like it should progress backwards with the largest asset being blue? (otherwise pink wins out)

<img width="443" alt="image" src="https://user-images.githubusercontent.com/7104689/197306353-08a4f859-03f4-42f0-b1fb-32c7ccabe29c.png">

Reply:
@MColl92 
Hmm I find the example confusing to reason about. In the original implementation, the palette was defined with colours mapping from largest to smallest footprint.
So if we call `#calculateFootprintPercent` on line 73, the colours will be mapped according to the index of already sorted categories.
https://github.com/KlimaDAO/klimadao/blob/0d2139db08c4563536332ba4f78a5e102baa908e/site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx#L70-L76

So it's redundant to sort again and calculate the palette index. We just need to ensure the colour palette defined is in the order we expect. I think it is worth revising the order of the colours, pink does look rather funny. It'd be much nicer to be on theme with green as the largest category.

Hope that makes sense!

Reply:
Yep I agree the logic is confusing. 

Since the sort on line 71 is from smallest to largest then no matter what order we define the `COLOR_PALLETTE` the "largest" (or `categories.length`) index value will always be variable (so no guarantee of the largest color).

For now I will reverse the array to ensure that smallest value is always green and we can revisit later?

Reply:
Let's wait for @theprofessora to make the call on how he'd like to apply the colour palette.
We've defined a few potential ways to approach it based on his call tho :D
IMO I don't think this is a showstopper though, we can always make a ticket and revisit this.

Reply:
Sure, happy to close and come back to this. Up to you guys 👍